tests/baselines/reference/tsbuildWatch/demo/updates-with-bad-reference.js

```diff
@@ -615,11 +615,11 @@ Output::
       "../../animals/index.ts": {
         "original": {
           "version": "-7220553464-import Animal from './animal';\n\nexport default Animal;\nimport { createDog, Dog } from './dog';\nexport { createDog, Dog };\n",
-          "signature": "1096904574-import Animal from './animal';\nexport default Animal;\nimport { createDog, Dog } from './dog';\nexport { createDog, Dog };\n",
+          "signature": "2198076214-export default Animal;\nexport { createDog, Dog };\n",
```

Comment:
This is first example of incorrect d.ts emit in the signature, Means when error is fixed, we will re-compile which is ok in this case i think. But would be great to check if we can preserve the behavior

Reply:
@weswigham  this one too if possible.

Reply:
Import links are collected by the `collectLinkedAliases` call in the emitter, so the condition on calling it needs to be toggled when you pass this into the emit resolver, as it is for `noCheck`.

Reply:
To elaborate, `emitDeclarationFileOrBundle` in `emitter.ts` needs to swap
``ts
        if ((emitOnly && !getEmitDeclarations(compilerOptions)) || compilerOptions.noCheck) {
``
to
```ts
        if ((emitOnly && !getEmitDeclarations(compilerOptions)) || compilerOptions.noCheck || forceDtsEmit) {
```

Reply:
It's pretty likely doing so will fix the other discrepancies you've seen here, since they all seem to do with alias elision.

Reply:
That did it. Thanks